tests/installation/addon_products_sle.pm

```diff
@@ -43,6 +43,19 @@ sub handle_all_packages_medium {
     # The legacy module is required if upgrade from previous version (bsc#1066338)
     push @addons, 'legacy' if get_var('UPGRADE') && !grep(/^legacy$/, @addons);
 
+    # For SLES12SPx and SLES11SPx to SLES15 migration, follow modules are need at least
+    # SLE-Module-Basesystem15
+    # SLE-Module-Containers15
+    # SLE-Module-Desktop-Applications15
+    # SLE-Module-Legacy15
+    # SLE-Module-Server-Applications15
+    # SLE-Module-Web-Scripting15
+    if (get_var('UPGRADE') && !sle_version_at_least('15', version_variable => 'HDDVERSION') && !check_var('SLE_PRODUCT', 'sled')) {
```

Comment:
@ldevulder @alvarocarvajald @asmorodskyi comments on this? Specifically the latest portion about !sled, maybe the same applies for other products?

Reply:
hard to say , it's related to upgrade only. and I have no information about SLE 12 SP3 with HPC module to SLE 15 HPC upgrade . what modules should / may be selected there. Not saying that it is normal , just stating the fact. And based on this statement - I am ok with this PR

Reply:
For `sles4sap` we need `base,desktop,serverapp,sapapp,ha`, but currently we don't have migration tests in place in o.s.d., so its inclusion into this part of the code can wait for the time being ... but if you don't mind adding it now, it will helps us in the long run.

Reply:
Same as @alvarocarvajald for HA, no upgrade tests in o.s.d. at this time. Should be added in the future.

Reply:
ok then we can keep it as is, no over-engineering without any tests

Reply:
No upgrade test for sles4sap product yet.
But HA extension is already covered in migration testing, for example:
media_upgrade_sles11sp4+ha+geo@64bit : https://openqa.suse.de/tests/1604474
media_upgrade_sles12sp3+ha+geo_allpatterns@64bit : https://openqa.suse.de/tests/1604110
proxyscc_upgrade_sles12sp3+ha+geo@64bit : https://openqa.suse.de/tests/1604102

You can find more upgrade tests for HA in SLE 15 job groups:
https://openqa.suse.de/group_overview/111
https://openqa.suse.de/group_overview/145

Reply:
@QingmingSu, thx! I will try to re-use these tests. The only think I have to do is to publish a HDD from these tests to after re-use it with all the HA stack tests.